### PR TITLE
PA-933 - [PDS-SystemTest-Automation]: Scale down portworx cluster

### PR DESCRIPTION
**Pull request by aravindranath14**
PA-219 - [PDS-SystemTest-Automation]: Scale up portworx cluster
PA-933 - [PDS-SystemTest-Automation]: Scale down portworx cluster

Branch test logs: 
https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds-branch-build-test/388/console

### CODE REVIEW

File: tests/pds/dataservice_test.go [hunk 1]
```diff
@@ -492,6 +492,132 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
 
 })
 
+var _ = Describe("{ScaleDownScaleupPXCluster}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("ScaleDownScaleupPXCluster", "Scales Down the PX cluster, Verify Data Services and Scales up the Px cluster", pdsLabels, 0)
+	})
+
+	It("Scale down PX pods, Verify Data Services and Scale up the PX pods", func() {
+		var generateWorkloads = make(map[string]string)
```
Comment (gauraa2) [inline tests/pds/dataservice_test.go:501]: Rather multiple times using var keyword , enclose all the variables with types.i.e
var (
x type_x
y type_y
)

File: tests/pds/dataservice_test.go [hunk 2]
```diff
@@ -492,6 +492,132 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+		var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+		var dsVersions = make(map[string]map[string][]string)
+		var deps []*pds.ModelsDeployment
+		var nodeList []*corev1.Node
+		torpedoNodeList := node.GetWorkerNodes()
```
Comment (gauraa2) [inline tests/pds/dataservice_test.go:506]: rename to just workerNode

File: tests/pds/dataservice_test.go [hunk 3]
```diff
@@ -492,6 +492,132 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+		Step("Deploy Data Services", func() {
+			for _, ds := range params.DataServiceToTest {
+				Step("Deploy and validate data service", func() {
+					isDeploymentsDeleted = false
```
Comment (gauraa2) [inline tests/pds/dataservice_test.go:510]: Not required

File: tests/pds/dataservice_test.go [hunk 4]
```diff
@@ -492,6 +492,132 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+					deployment, _, dataServiceVersionBuildMap, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+					log.FailOnError(err, "Error while deploying data services")
+					deployments[ds] = deployment
+					dsVersions[ds.Name] = dataServiceVersionBuildMap
+					nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+					log.FailOnError(err, "Error while getting Data Serice Nodes")
+					myNode := nodes[0]
```
Comment (gauraa2) [inline tests/pds/dataservice_test.go:517]: rename it i.e dataServiceDeploymentNode ...

File: tests/pds/dataservice_test.go [hunk 5]
```diff
@@ -492,6 +492,132 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+					var params pdslib.WorkloadGenerationParams
+					pod, dep, err = RunWorkloads(params, ds, deployment, namespace)
+					log.FailOnError(err, fmt.Sprintf("Error while genearating workloads for dataservice [%s]", ds.Name))
+					if dep == nil {
+						generateWorkloads[ds.Name] = pod.Name
+					} else {
+						generateWorkloads[ds.Name] = dep.Name
+					}
+					for dsName, workloadContainer := range generateWorkloads {
+						log.Debugf("dsName %s, workloadContainer %s", dsName, workloadContainer)
```
Comment (gauraa2) [inline tests/pds/dataservice_test.go:545]: log it as Info

**Review by gauraa2 — CHANGES_REQUESTED** [review]
Please attach the logs.

File: tests/pds/dataservice_test.go (hunk 4 above)
Comment (aravindranath14) [reply tests/pds/dataservice_test.go]: Done

File: tests/pds/dataservice_test.go (hunk 5 above)
Comment (aravindranath14) [reply tests/pds/dataservice_test.go]: Done

**Review by gauraa2 — APPROVED** [review]
LGTM

File: tests/pds/dataservice_test.go [hunk 6]
```diff
@@ -492,6 +492,133 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+					var params pdslib.WorkloadGenerationParams
+					pod, dep, err = RunWorkloads(params, ds, deployment, namespace)
+					log.FailOnError(err, fmt.Sprintf("Error while genearating workloads for dataservice [%s]", ds.Name))
+					if dep == nil {
+						generateWorkloads[ds.Name] = pod.Name
+					} else {
+						generateWorkloads[ds.Name] = dep.Name
+					}
+					for dsName, workloadContainer := range generateWorkloads {
+						log.Info("dsName %s, workloadContainer %s", dsName, workloadContainer)
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:546]: change it to Infof

File: tests/pds/dataservice_test.go [hunk 7]
```diff
@@ -492,6 +492,133 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+			log.InfoD("Successfully Scaled up Px Pods")
+		})
+		log.InfoD("Sleeping 60 seconds again after scale up of PX pods before we check the health of Data Services")
+		time.Sleep(60 * time.Second)
+		Step("Verify the Data Services status status after scaling up PX pods", func() {
+			for _, dep := range deps {
+				err = dsTest.ValidateDataServiceDeployment(dep, namespace)
+				log.FailOnError(err, "Error while validating data services")
+			}
+			log.InfoD("Deployments pods are up and healthy after scaleup PX Pods")
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:614]: nit: add a space after scale

File: tests/pds/dataservice_test.go [hunk 8]
```diff
@@ -492,6 +492,133 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+						log.FailOnError(err, "Failed to stop PX on Node %v", workerNode.Name)
+						break
+					}
+				}
+			}
+			log.InfoD("Successfully Scaled Down PX Pods...")
+		})
+		log.InfoD("Sleeping 60 seconds after scale down of PX pods before we check the health of Data Services")
+		time.Sleep(60 * time.Second)
+		Step("Verify the Data Services status status after scaling down the PX pods", func() {
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:585]: remove one status

File: tests/pds/dataservice_test.go [hunk 9]
```diff
@@ -492,6 +492,133 @@ var _ = Describe("{EnableandDisableNamespace}", func() {
...
+					} else if Contains(dataServicePodWorkloads, dsName) {
+						log.InfoD("Deleting Workload Generating pod %v ", workloadContainer)
+						err = pdslib.DeleteK8sPods(workloadContainer, namespace)
+					}
+					log.FailOnError(err, "error deleting workload generating pods")
+				})
+			}
+		}()
+
+		Step("Scaledown PX by Labelling Nodes on which Data Services are running", func() {
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:567]: Add a space after scale

**Review by lsrinivas-pure — COMMENTED** [review]
just a minor comments

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**